### DISCUSSION TITLE
How to tell what CPU extension implementation of PQC the liboqs is using(AVX2 / Ref) ?

### DISCUSSION THREAD

**Discussion by wadaphaq** [open-quantum-safe/liboqs#1895] [Q&A] [answered]
Hello,
Upon inspection, I noticed that after building the liboqs, I noticed that both the reference version and the CPU extension-enabled version of the pqc are built ie, avx2 and ref or aarch64 and ref in the build folder. Is there any way to specify what implementation of the pqc the liboqs, and by extension, the oqs provider use?

**Comment by baentsch** [comment, marked as answer]
Yes. Please read up on the [OQS_DIST_BUILD](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#oqs_dist_build) option.

**Reply by wadaphaq** [reply]
Hello, thank you for your answer, but I am still a little confused. After building the library I can still find folders such as $algo_avx2 or $algo_aarch64, doesn't matter whether the library is build with DIST_BUILD on or off. Is there any ways to specifically use the ref version of the algorithm or aarch64 version when testing?

**Reply by SWilson4** [reply]
If you want to use the `ref` implementation, set [`OQS_DIST_BUILD`](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#oqs_dist_build) to `OFF` and [`OQS_OPT_TARGET`](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#oqs_opt_target) to `generic`.
